system/Entity.php

```diff
@@ -427,6 +427,19 @@ public function __isset(string $key): bool
 		return isset($this->attributes[$key]);
 	}
 
+	/**
+	 * Set raw data array without any mutations
+	 *
+	 * @param  array $data
+	 * @return $this
+	 */
+	public function setRawArray(array $data)
```

Comment:
Rename to `setAttributes` please.

Reply:
What checks do you mean?

Reply:
Not a fan of the name change - it's not an abstract class. Even then not sure i'd like it.

Was referring to the checks in the database layer that you've added where it checks if it's an Entity and then uses `setRawArray`. If we move that to the Entity constructor I don't think any of those checks are needed since things will be set to their raw values the way the database layer currently works.

Reply:
Then we say goodbye to the stdClass class and the custom class.
Because stdClass has no constructor, and the custom class may not have a constructor.

Reply:
Nothing's changing there. It works currently. The database engine itself handles that in `$this->resultID->fetch_object($className);`. 

I'm not saying you should remove existing functionality - but remove the portions you added in this PR. They all look something like: 

```
if (is_subclass_of($className, Entity::class))
		{
			$data = $this->fetchAssoc();
			return empty($data) ? false : (new $className())->setRawArray($data);
		}
```

Reply:
I'm confused. 
If I remove this code
```php
if (is_subclass_of($className, Entity::class))
{
    return empty($data = $this->fetchAssoc()) ? false : new $className($data);
}
```
then this PR does not make sense.

Reply:
Actually - I might be dense here. I was thinking that it would set through the constructor, but it doesn't. It tries to set the value directly on the object. It's late and I'm tired and distracted with other problems. Sorry - what you've done works fine. Just rename that method, please.